src/platform/stub-setup.ts

```diff
@@ -4,25 +4,108 @@
+import {DeviceManagerStub} from './device-manager';
 
 export const smarthomeStub: {
   App: typeof smarthome.App;
   Intents: {[key in keyof typeof smarthome.Intents]: string};
+  DataFlow: {
+    UdpRequestData: typeof smarthome.DataFlow.UdpRequestData;
+  };
+  Constants: {
+    Protocol: {[key in keyof typeof smarthome.Constants.Protocol]: string};
+  };
+  Execute: typeof smarthome.Execute;
 } = {
   App: AppStub,
   Intents: {
     EXECUTE: 'action.devices.EXECUTE',
     IDENTIFY: 'action.devices.IDENTIFY',
     REACHABLE_DEVICES: 'action.devices.REACHABLE_DEVICES',
   },
+  DataFlow: {
+    UdpRequestData: class {
+      data!: string;
+      requestId!: string;
+      deviceId!: string;
+      protocol: smarthome.Constants.Protocol = smarthome.Constants.Protocol.UDP;
+      port!: number;
+    },
+  },
+  Constants: {
+    Protocol: {
+      BLE: 'BLE',
+      HTTP: 'HTTP',
+      TCP: 'TCP',
+      UDP: 'UDP',
+      BLE_MESH: 'BLE_MESH',
+    },
+  },
+  Execute: {
+    Response: {
+      Builder: class {
+        private requestId: string = '';
+        private commands: smarthome.IntentFlow.ExecuteResponseCommands[] = [];
+        public setRequestId(requestId: string): this {
+          this.requestId = requestId;
+          return this;
+        }
+        public setSuccessState(deviceId: string, state: unknown): this {
+          this.commands.push({
+            ids: [deviceId],
+            status: 'SUCCESS',
+            states: state,
+          });
+          return this;
+        }
+        public setErrorState(
+          deviceId: string,
+          errorCode: smarthome.IntentFlow.ExecuteErrors
+        ): this {
+          this.commands.push({
+            ids: [deviceId],
+            status: 'ERROR',
+            errorCode,
+          });
+          return this;
+        }
+        public build() {
+          return {
+            requestId: this.requestId,
+            payload: {
+              commands: this.commands,
+            },
+          };
+        }
+      },
+    },
+  },
 };
 
-// Promotes App to AppStub
+/**
+ * Module-level function to source a `MockLocalHomePlatform` from an `App`.
+ * @param app  The app to promote and extract the `MockLocalHomePlatform` from.
+ * @returns  The `MockLocalHomePlatform` member.
+ */
 export function extractMockLocalHomePlatform(
   app: smarthome.App
-): MockLocalHomePlatform{
+): MockLocalHomePlatform {
   if (app instanceof AppStub) {
     return app.getLocalHomePlatform();
   }
   throw new Error("Couldn't downcast App to AppStub");
 }
+/**
+ * Module-level function to source a `DeviceManagerStub` from an `App`.
+ * @param app  The app to promote and extract the `DeviceManagerStub` from.
+ * @returns  The `DeviceManagerStub` member.
+ */
+export function extractDeviceManagerStub(
+  app: smarthome.App
+): DeviceManagerStub {
+  const deviceManager = app.getDeviceManager();
+  if (deviceManager instanceof DeviceManagerStub) {
```

Comment:
There's no chance this won't be true.  We could remove these functions, but then the developer would need to manually downcast their `App` to `AppStub` in every test and mute the `eslint` error every time they want to use the stubs.  These functions also help hide `AppStub` from ever being used in a developer test.

The other option is to require the developer to use explicitly construct `AppStub` objects.  They would then have direct access to the stubs.  This is probably what I'll end up doing, since the developer is using other stubs explicitly as well.

@proppy also suggested removing these functions.

Reply:
maybe you could provide both with a single top-module function that return an interface?

Reply:
Done